cmd/jujud/agent/machine.go

```diff
@@ -900,9 +900,13 @@ func (a *MachineAgent) setupContainerSupport(st api.Connection, logger machine.L
 	m := result[0].Machine
 
 	var supportedContainers []instance.ContainerType
-	supportsContainers := container.ContainersSupported()
-	if supportsContainers {
-		supportedContainers = append(supportedContainers, instance.LXD)
+	supportedContainerType, err := container.ContainersSupported()
+	if err != nil {
+		logger.Warningf("determining container support: %v\nno containers possible", err)
+	}
+
+	if err == nil && supportedContainerType != "" {
+		supportedContainers = append(supportedContainers, supportedContainerType)
 	}
```

Comment:
Can we just delete this and tell me what the consequences are?

I suspect this has been broken, so it was never correctly filled in and we haven't noticed till I removed KVM. Considering that, just remove it and try the following:

 - `juju bootstrap lxd`
 - `juju bootstrap aws` and `juju add-machine lxd`
 - `juju deploy ubuntu --to lxd`

This should cover the common cases to ensure this works (there are more)

Reply:
You're right in this suspicion, removed all of these checks (including the kvm check) and everything seems to be working just fine (on both lxd and aws).